.travis.yml

```diff
@@ -98,6 +98,8 @@ install:
 script: |
   set -e
   if [[ $JOB = docs ]]; then
+    # install docs dependencies
+    pip install '.[docs]'
```

Comment:
If I understand the syntax for this `pip install .[whatever]` correctly then it is just a dict lookup in python. So we could add "alot" of names for all kind of dependency sets. But to add more special entries to the extra_requires dict does only make sense if that are groups of dependencies that we want to expose to users/package maintainers/...  So I would only do it for obvious things like "docs" and "tests".

IMHO The line 77 that you mention is something that is specific to our travis workflow. We only want to build the docs at this point and do not care to really run alot. Therefor we could actually mock all its dependencies by writing dummy python files. But then we would have to take care to define some empty classes and functions in these files which are imported by name from within alot. My approach was to mock the modules with annoying C dependencies which are not easily available on travis (compare line 78) and install all the other dependencies normally (but list them by hand because notmuch and gpgme are missing from the list). 

We could merge line 77 into line 84 though.